registration/forms.py

```diff
@@ -40,6 +40,20 @@ class Meta:
         fields = (UsernameField(), "email")
 
 
+class RegistrationFormUsernameLowercase(RegistrationForm):
+    """
+    A subclass of :class:`RegistrationForm` which enforces unique case insensitive
+    usernames, make all usernames to lower case.
+
+    """
+    def clean_username(self):
+        username = self.cleaned_data.get('username', '').lower()
+        if User.objects.filter(username=username).exists():
```

Comment:
Ok, I found it in stackoverflow: http://stackoverflow.com/questions/4720079/django-query-filter-with-variable-column.
It could be finished.